### PR TITLE
Minor journey fixes

### PR DESCRIPTION
**Pull request by cp-radhika-s**
## Summary by CodeRabbit

- **New Features**
	- Enhanced `.gitignore` to ignore specific Flutter plugin files and configuration files.
	- Added permission for foreground location services in the Android manifest.
  
- **Bug Fixes**
	- Improved error handling and clarity in location-saving methods within the journey repository.

- **Chores**
	- Removed obsolete files related to Flutter plugin configurations and a simple calculator class.
  
- **Refactor**
	- Simplified method signatures in the `JourneyRepository` class for better readability and maintainability.
	- Streamlined the `fetchCurrentLocation` method in the `HomeViewNotifier` for improved clarity.
	- Consolidated location management logic in the `LocationManager` class for enhanced functionality.
	- Updated bitmap descriptor creation methods for better consistency in the map handling classes.

### CODE REVIEW

File: data/lib/repository/journey_repository.dart [hunk 1]
```diff
@@ -234,7 +257,7 @@ class JourneyRepository {
       toLatitude: extractedLocation.latitude,
       toLongitude: extractedLocation.longitude,
       routeDistance: distance,
-      routeDuration: null,
+      routeDuration: duration,
```
Comment (cp-ishita-g) [inline data/lib/repository/journey_repository.dart:260]: we don't required duration here we are showing duration only for steady journey

File: data/lib/repository/journey_repository.dart [hunk 2]
```diff
@@ -296,7 +328,13 @@ class JourneyRepository {
       route_distance: distance + (lastKnownJourney.route_distance ?? 0),
       route_duration: (lastKnownJourney.update_at ?? 0) -
           (lastKnownJourney.created_at ?? 0),
-      routes: _getRoute(userId),
```
Comment (cp-ishita-g) [inline data/lib/repository/journey_repository.dart:299]: remove this function from below, if we are not going to use this in future

File: data/lib/repository/journey_repository.dart (hunk 1 above)
Comment (cp-radhika-s) [reply data/lib/repository/journey_repository.dart:260]: It's fine, let's keep it this change

File: data/lib/repository/journey_repository.dart [hunk 3]
```diff
@@ -271,18 +296,25 @@ class JourneyRepository {
...
+      routes: lastKnownJourney.routes +
+          [
+            JourneyRoute(
+              latitude: extractedLocation.latitude,
+              longitude: extractedLocation.longitude,
+            )
+          ],
```
Comment (cp-ishita-g) [inline data/lib/repository/journey_repository.dart:305]: why did you make this change above one also working with expected case

File: data/lib/repository/journey_repository.dart [hunk 4]
```diff
@@ -29,14 +30,16 @@ class JourneyRepository {
   Future<void> saveLocationJourney({
     required LocationData extractedLocation,
     required String userId,
-    bool? fromHomeViewModel,
```
Comment (cp-ishita-g) [inline data/lib/repository/journey_repository.dart:32]: two same location are added on day change, to fix that i used this way, why did you remove it, did you test day change journey?

File: data/lib/repository/journey_repository.dart (hunk 3 above)
Comment (cp-radhika-s) [reply data/lib/repository/journey_repository.dart:305]: We were setting routes from the last five cached locations. but it should be  
`lastKnownJourney` routes + new location

File: data/lib/repository/journey_repository.dart (hunk 4 above)
Comment (cp-radhika-s) [reply data/lib/repository/journey_repository.dart:32]: we were not using this param at all, it was always false

File: data/lib/repository/journey_repository.dart (hunk 3 above)
Comment (cp-ishita-g) [reply data/lib/repository/journey_repository.dart:305]: no it's not last five cached location it's all the location we are getting when user was moving from location a to location b

File: data/lib/repository/journey_repository.dart (hunk 4 above)
Comment (cp-ishita-g) [reply data/lib/repository/journey_repository.dart:32]: it is passed from home view model,

Comment (cp-radhika-s) [reply data/lib/repository/journey_repository.dart:32]: We shouldn't update the journey from the home screen. I wasn't sure why we were making journey calls from the home screen.

File: data/lib/repository/journey_repository.dart (hunk 3 above)
Comment (cp-radhika-s) [reply data/lib/repository/journey_repository.dart:305]: We have a cache limit of 200, what if it exceeds it?

File: data/lib/repository/journey_repository.dart (hunk 4 above)
Comment (cp-ishita-g) [reply data/lib/repository/journey_repository.dart:32]: to fix one issue like we we sign out or delete account and that again login with out deleting app form same location user was, than we are not getting current location, as delegate method will give location when there is location update, so we don't get location update it is showing location not available, and other method were also not called to start a journey

Comment (cp-radhika-s) [reply data/lib/repository/journey_repository.dart:32]: okay, it's auth related issue so it should be handled separately not from the home screen. I'll fix it

Comment (cp-ishita-g) [reply data/lib/repository/journey_repository.dart:32]: no it is not auth related issue, just location update method is not getting called as we don't have location update from last location which was saved, we need to request to get current location and save and base on that location move to journey logic which to add

Comment (cp-radhika-s) [reply data/lib/repository/journey_repository.dart:34]: Location and journey should be updated from the location update stream only. In the above case, we should stop tracking on logout and start again when the user logs in.

Comment (cp-ishita-g) [reply data/lib/repository/journey_repository.dart:34]: we get location update only when there is change in location from last location

File: app/lib/ui/flow/journey/timeline/journey_timeline_screen.dart [hunk 5]
```diff
@@ -178,9 +178,11 @@ class _JourneyTimelineScreenState extends ConsumerState<JourneyTimelineScreen> {
     final location = LatLng(journey.from_latitude, journey.from_longitude);
     final steadyDuration =
         notifier.getSteadyDuration(journey.created_at!, journey.update_at!);
-    final formattedTime = (isFirstItem)
-        ? _getFormattedLocationTimeForFirstItem(journey.created_at!)
-        : _getFormattedTimeForSteadyLocation(journey.created_at!);
+    final formattedTime = _getFormattedJourneyTime(
+        journey.created_at ?? 0, journey.update_at ?? 0);
+    // final formattedTime = (isFirstItem)
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/journey/timeline/journey_timeline_screen.dart:183]: Remove unnecessary code

**Pull request merged**